File: ABCtask1/triangle.cpp
#include <fstream>
#include <cmath>
#include <cstring>
#include "triangle.h"
#include <iostream>
int Triangle::In(FILE *input_file)
{
    char color_type[10001];
    int res = fscanf(input_file, "%d %d %d %d %d %d %10000s", &x1, &y1, &x2, &y2, &x3, &y3, color_type);
    if (res != 7)
    {
        std::cout<<x1<<y1<<std::endl;
        printf("Could not read string\n");
        return 1;
    }
    color = GetColor(color_type);
    return 0;
}

void Triangle::Out(FILE *output_file)
{
    printf("Triangle: (%d; %d); (%d; %d); (%d; %d); Color = %s;\n Perimeter = %lf \n",
          x1, y1, x2, y2, x3, y3, ColorToString(color).c_str(), Perimeter());
    fprintf(output_file, "Triangle: (%d; %d); (%d; %d); (%d; %d); Color = %s;\n Perimeter = %lf \n",
            x1, y1, x2, y2, x3, y3, ColorToString(color).c_str(), Perimeter());
}

double Triangle::Perimeter()
{
    return Side(x1, y1, x2, y2) + Side(x1, y1, x3, y3) + Side(x2, y2, x3, y3);
}

double Triangle::Side(int x1, int y1, int x2, int y2)
{
    return sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
}
std::string Triangle::ColorToString(Color color)
{
    switch (color)
    {
    case RED:
        return "RED";
    case ORANGE:
        return "ORANGE";
    case YELLOW:
        return "YELLOW";
    case GREEN:
        return "GREEN";
    case LIGHT_BLUE:
        return "LIGHT_BLUE";
    case BLUE:
        return "BLUE";
    case VIOLET:
        return "VIOLET";
    default:
        return "NOT A COLOR!";
    }
}

Triangle::Color Triangle::GetColor(char color_type[10001])
{
    if (strcmp(color_type, "RED") == 0)
    {
        return RED;
    }
    if (strcmp(color_type, "ORANGE") == 0)
    {
        return ORANGE;
    }
    if (strcmp(color_type, "YELLOW") == 0)
    {
        return YELLOW;
    }
    if (strcmp(color_type, "GREEN") == 0)
    {
        return GREEN;
    }
    if (strcmp(color_type, "LIGHT_BLUE") == 0)
    {
        return LIGHT_BLUE;
    }
    if (strcmp(color_type, "BLUE") == 0)
    {
        return BLUE;
    }
    if (strcmp(color_type, "VIOLET") == 0)
    {
        return VIOLET;
    }
    return VIOLET;
}

File: pythonProject/triangle.py
import math
import random


# Class of triangle
class Triangle:
    # Constructor
    def __init__(self):
        self.x1 = 0
        self.y1 = 0
        self.x2 = 0
        self.y2 = 0
        self.x3 = 0
        self.y3 = 0
        self.color = ""
        self.perimeter = 0.0

    # Inputting triangle data from existing file
    def triangle_in(self, params):
        if len(params) != 8:
            print("Could not read string")
            return 1
        else:
            self.x1 = int(params[1])
            self.y1 = int(params[2])
            self.x2 = int(params[3])
            self.y2 = int(params[4])
            self.x3 = int(params[5])
            self.y3 = int(params[6])
            self.color = params[7]
            self.perimeter = self.side(self.x1, self.y1, self.x2, self.y2) + \
                             self.side(self.x1, self.y1, self.x3, self.y3) + \
                             self.side(self.x2, self.y2, self.x3, self.y3)

    # Getting random parameters for triangle
    def triangle_in_rnd(self):
        while ((self.side(self.x1, self.y1, self.x2, self.y2) >= self.side(self.x1, self.y1, self.x3, self.y3) +
               self.side(self.x2, self.y2, self.x3, self.y3)) or (self.side(self.x1, self.y1, self.x3, self.y3)
               >= self.side(self.x1, self.y1, self.x2, self.y2) + self.side(self.x2, self.y2, self.x3, self.y3)) or
               (self.side(self.x2, self.y2, self.x3, self.y3) >= self.side(self.x1, self.y1, self.x2, self.y2) +
               self.side(self.x1, self.y1, self.x3, self.y3))):
            self.x1 = random.randint(1, 1000)
            self.y1 = random.randint(1, 1000)
            self.x2 = random.randint(1, 1000)
            self.y2 = random.randint(1, 1000)
            self.x3 = random.randint(1, 1000)
            self.y3 = random.randint(1, 1000)
        self.get_color()
        self.perimeter = self.side(self.x1, self.y1, self.x2, self.y2) + \
                             self.side(self.x1, self.y1, self.x3, self.y3) + \
                             self.side(self.x2, self.y2, self.x3, self.y3)

    # Getting a side of triangle
    @staticmethod
    def side(x1, y1, x2, y2):
        return math.sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2))

    # Outputting rectangle data to console and output file
    def out(self, file):
        output = ("Triangle: (" + str(self.x1) + "; " + str(self.y1) + "), (" + str(self.x2) + "; " + str(self.y2) + "); (" +
                  str(self.x3) + "; " + str(self.y3) + "); Color = " + self.color +
                  "Perimeter = " + str(self.perimeter))
        file.write(output)
        print(output)

    # Random generation for triangle color
    def get_color(self):
        color_int = random.randint(0, 6)
        if color_int == 0:
            self.color = "RED\n"
        if color_int == 1:
            self.color = "ORANGE\n"
        if color_int == 2:
            self.color = "YELLOW\n"
        if color_int == 3:
            self.color = "GREEN\n"
        if color_int == 4:
            self.color = "LIGHT_BLUE\n"
        if color_int == 5:
            self.color = "BLUE\n"
        if color_int == 6:
            self.color = "VIOLET\n"
        return 0

